common/main/cpp/native_c4document.cc

```diff
@@ -151,6 +151,58 @@ Java_com_couchbase_lite_internal_core_impl_NativeC4Document_getSelectedRevID(JNI
     return toJString(env, doc->selectedRev.revID);
 }
 
+/*
+ * Class:     com_couchbase_lite_internal_core_impl_NativeC4Document
+ * Method:    getRevisionHistory
+ * Signature: (JJJ[Ljava/lang/String;)Ljava/lang/String;
+ */
+JNIEXPORT jstring JNICALL
+Java_com_couchbase_lite_internal_core_impl_NativeC4Document_getRevisionHistory(
+        JNIEnv *env,
+        jclass ignore,
+        jlong jcoll,
+        jlong jdoc,
+        jlong maxRevs,
+        jobjectArray jBackToRevs) {
+    C4String *backToRevs = nullptr;
+    unsigned int nBackToRevs = 0;
+
+    // if jBackToRevs is non-null, a Java String[], convert it to a C array of C4Slice:
+    if (jBackToRevs != nullptr) {
+        jsize n = env->GetArrayLength(jBackToRevs);
+        if (env->EnsureLocalCapacity(std::min(n + 1, MaxLocalRefsToUse)) < 0)
+            return nullptr;
```

Comment:
Not sure if this check is needed as I don't think we create any new references to java objects? I might misunderstood this function though.

Reply:
You are absolutely right.  Removing it.